digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -1196,21 +1193,21 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
       halt(Unauthorized("ServiceRoad user is only authorized to alter serviceroad assets"))
     val asset = (parsedBody \ "asset").extract[service.IncomingAsset]
 
-//    for (link <- roadLinkService.fetchVVHRoadlinkAndComplementary(asset.linkId)) {
     for (link <- roadLinkService.getRoadLinkAndComplementaryFromVVH(asset.linkId)) {
       validateUserMunicipalityAccess(user)(link.municipalityCode)
       optTypeID match {
         case Some(typeId) => validateAdministrativeClass(typeId)(link.administrativeClass)
         case _ => None
       }
-//      service.create(asset, user.username, link.geometry, link.municipalityCode, Some(link.administrativeClass), link.linkSource)
       service.create(asset, user.username, link)
     }
   }
 
-  private def verifyServiceToUse(typeId: Int): LinearAssetOperations = {
+  private def getLinearAssetService(typeId: Int): LinearAssetOperations = {
     typeId match {
-      case maintenanceRoadService.maintenanceRoadAssetTypeId => maintenanceRoadService
+      case LinearAssetTypes.MaintenanceRoadAssetTypeId => maintenanceRoadService

```

Comment:
It could be used the values defined on asset.scala instead of the LinearAssetTypes